### PR TITLE
[REG-1965] Handle AI Service auth tokens

### PR DESCRIPTION
**Pull request by RG-nAmKcAz**
- Passes auth tokens to AI Service
- Refactors service manager patterns to singletons to remove monobehaviour links and remove warnings when using the settings menus
- Updates prefabs to remove behaviour references
- Refactors/CleansUp CV naming to be AI

---

Find the pull request instructions [here](https://www.notion.so/regressiongg/Pull-Request-Process-0d57a7eb582a446983edfacafa406f1e?pvs=4)

Every reviewer and the owner of the PR should consider these points in their request (feel free to copy this checklist so you can fill it out yourself in the overall PR comment)

- [ ] The code is extensible and backward compatible
- [ ] New public interfaces are extensible and open to backward compatibility in the future
- [ ] If preparing to remove a field in the future (i.e. this PR removes an argument), the argument stays but is no longer functional, and attaches a deprecation warning. A linear task is also created to track this deletion task.
- [ ] Non-critical or potentially modifiable arguments are optional
- [ ] Breaking changes and the approach to handling them have been verified with the team (in the Linear task, design doc, or PR itself)
- [ ] The code is easy to read
- [ ] Unit tests are added for expected and edge cases
- [ ] Integration tests are added for expected and edge cases
- [ ] Functions and classes are documented
- [ ] Migrations for both up and down operations are completed
- [ ] A documentation PR is created and being reviewed for anything in this PR that requires knowledge to use
- [ ] Implications on other dependent code (i.e. sample games and sample bots) is considered, mentioned, and properly handled
- [ ] Style changes and other non-blocking changes are marked as non-blocking from reviewers

### CODE REVIEW

File: src/gg.regression.unity.bots/Runtime/Scripts/RGServiceManager.cs [hunk 1]
```diff
@@ -49,7 +40,7 @@ public bool IsAuthed()
             return _rgAuthToken != null;
         }
 
-        public bool LoadAuth()
```
Comment (addisonbgross) [inline src/gg.regression.unity.bots/Runtime/Scripts/RGServiceManager.cs:52]: Nit: It could be nice to have an overloaded use of this method, to match how it gets used a bunch below. Like:
```
public bool LoadAuth()
{
    return LoadAuth(out _);
}
```

**Review by addisonbgross — APPROVED** [review]
Aside from one tiny nit, this LGTM

**Review by abeizer — APPROVED** [review]

File: src/gg.regression.unity.bots/Runtime/Prefabs/RGOverlayCanvas.prefab [hunk 2]
```diff
@@ -2893,7 +2893,6 @@ GameObject:
   - component: {fileID: 2671455581720235337}
```
Comment (vontell) [inline src/gg.regression.unity.bots/Runtime/Prefabs/RGOverlayCanvas.prefab:2893]: Thank you for updating both prefabs!

File: src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/AIServiceManager.cs [hunk 3]
```diff
@@ -51,98 +53,110 @@ private String GetCvServiceBaseUri()
...
-                    var response = JsonConvert.DeserializeObject<CVImageResultList>(s);
-                    onSuccess.Invoke(response.results);
-                },
-                onFailure: (f) =>
-                {
-                    RGDebug.LogWarning($"Failed to evaluate CV Image criteria: {f}");
-                    onFailure.Invoke();
-                }
-            );
+            if (RGServiceManager.GetInstance().LoadAuth(out var authToken))
```
Comment (vontell) [inline src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/AIServiceManager.cs:56]: If they are not authed, what happens? It seems like neither action gets called which could cause hanging somewhere? Maybe we should either throw an exception or we could call onFailure. But at a minimum I'd like the user to at least see a message somewhere that they can't use the AI API without being logged in.

**Review by vontell — CHANGES_REQUESTED** [review]
Nice! Just a small note about error messages for the case of not being logged in.

File: src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/AIServiceManager.cs (hunk 3 above)
Comment (RG-nAmKcAz) [reply src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/AIServiceManager.cs:56]: LoadAuth internally logs a warning if the APIKey isn't configured, same as it does for the existing RGService calls.  If an invalid key is configured, you will get 401 errors from the AIserver calls.

**Review by vontell — APPROVED** [review]
